### ISSUE TITLE
UI causes Firefox to use 30-60% of CPU

### ISSUE THREAD

**Issue by erikschul** [argoproj/argo-cd#21053]
ArgoCD version: v2.12.6

**Describe the bug**

`/applications?showFavorites=false&proj=&sync=&autoSync=&health=&namespace=&cluster=&labels=`

<img width="588" alt="image" src="https://github.com/user-attachments/assets/449f4320-c7cf-44d6-b793-ba41d907bc72">

I frequently have to close the tab just to avoid wasting CPU.

But it's not all the time. So maybe it sometimes gets stuck on something?

This is a complicated bug to debug, but I thought you should know.

**Comment by andrii-korotkov-verkada** [comment]
How many applications do you have?

**Comment by erikschul** [comment]
Hmm, 25 applications, 1.7k resource objects, 119 api resources
according to Grafana

**Comment by andrii-korotkov-verkada** [comment]
Yeah, that's not much. Are you able to get some profiling data about which code consumes the CPU?

**Comment by erikschul** [comment]
No longer using ArgoCD.
